src/strkey.js

```diff
@@ -70,14 +54,13 @@ export function encodeBase58Check(versionByteName, data) {
   let checksum      = calculateChecksum(payload);
   let unencoded     = Buffer.concat([payload, checksum]);
 
-  return encodeBase58(unencoded);
+  return base32.encode(unencoded).toString();
 }
 
 function calculateChecksum(payload) {
-  let inner = hash(payload);
-  let outer = hash(inner);
-  return outer.slice(0,4);
-} 
+  let checksum = ("0000"+crc.crc16xmodem(payload).toString(16)).slice(-4); // Padding to 2 bytes
```

Comment:
You could provide some more explanatory text around this block? it's not clear what your algorithm is and why.  For example:

- What  are you padding to 2 bytes?  what is it's length prior?  Why is it being padded to two bytes?
- Are you converting a big-endian value to little-endian, or is it the other way around?  The code doesn't specify, it's just flipping endianness.